tls-client/main.cpp

```diff
@@ -224,15 +219,25 @@ class HelloHTTPS {
                 ret == MBEDTLS_ERR_SSL_WANT_WRITE));
         if (ret < 0) {
             print_mbedtls_error("mbedtls_ssl_handshake", ret);
+            _tcpsocket->close();
             return;
         }
 
+        /* Fill the request buffer */
+        _bpos = snprintf(_buffer, sizeof(_buffer) - 1, "GET %s HTTP/1.1\nHost: %s\n\n", path, HTTPS_SERVER_NAME);
+        
+        int written = 0;
         do {
             ret = mbedtls_ssl_write(&_ssl, (const unsigned char *) _buffer, _bpos);
-        } while (ret != 0 && (ret == MBEDTLS_ERR_SSL_WANT_READ ||
-                ret == MBEDTLS_ERR_SSL_WANT_WRITE));
+            if (ret > 0)
+              written += ret;
+            else if (ret != MBEDTLS_ERR_SSL_WANT_READ &&
```

Comment:
Because this condition is not sufficient as the loop condition. ```ret``` can have positive values. But I can squeeze the new condition in while if you want.